pilot/pkg/bootstrap/server.go

```diff
@@ -1249,3 +1262,22 @@ func (s *Server) addFileWatcher(file string, callback func()) {
 		}
 	}()
 }
+
+func (s *Server) waitForCacheSync(stop <-chan struct{}) bool {
+	if !cache.WaitForCacheSync(stop, func() bool {
```

Comment:
AFAIK that's just a busy wait calling that func ?

The plan is to remove the deps on k8s ( galley will abstract it), any chance we can just inline that small code here, it's just a simple loop - no need to add an extra dep, and there is really no cache involved, just waiting for MCP to load.

Reply:
Yes, we can do that in 1.2 I think.

Reply:
Wont this break other platforms where there is no k8s? like Consul or CloudFoundry that runs pilot without k8s?